src/VisualStudio/Core/Def/ValueTracking/ValueTrackedTreeItemViewModel.cs

```diff
@@ -82,28 +76,35 @@ private void CalculateChildren()
                 return;
             }
 
+            TreeViewModel.LoadingCount++;
             _childrenCalculated = true;
 
-            CalculateChildrenAsync(CancellationToken.None)
-                 .ContinueWith(task =>
-                 {
-                     if (task.Exception is not null)
-                     {
-                         _childrenCalculated = false;
-                         return;
-                     }
-
-                     ChildItems.Clear();
-
-                     foreach (var item in task.Result)
-                     {
-                         ChildItems.Add(item);
-                     }
-                 },
-
-                 // Use the UI thread synchronization context for calling back to the UI
-                 // thread to add the tiems
-                 TaskScheduler.FromCurrentSynchronizationContext());
+            System.Threading.Tasks.Task.Run(async () =>
+            {
+                try
+                {
+                    var items = await _valueTrackingService.TrackValueSourceAsync(_trackedItem, ThreadingContext.DisposalToken).ConfigureAwait(false);
+
+                    await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    ChildItems.Clear();
+
+                    foreach (var valueTrackedItem in items)
+                    {
+                        ChildItems.Add(new ValueTrackedTreeItemViewModel(
+                                valueTrackedItem,
+                                _solution,
+                                TreeViewModel,
+                                _glyphService,
+                                _valueTrackingService,
+                                ThreadingContext));
+                    }
+                }
+                finally
+                {
+                    await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    TreeViewModel.LoadingCount--;
```

Comment:
Emm, is this value thread-safe?

Reply:
It's always modified on the UI thread... in fact it must be or WPF will throw because it triggers a binding

Reply:
It's a UI bound property, so it must always be modified on the UI thread. WPF will throw otherwise. Everywhere it is modified requires a `SwitchToMainThreadAsync` or `ThrowIfNotOnMainThread`